### PR TITLE
Allow psr/http-message v2

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Instead of #223 to not bump to PHP 7.2 when not needed.

Requires https://github.com/php-http/psr7-integration-tests/pull/68

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
PR is ready, tests will still pass after php-http/psr7-integration-tests#68 is merged.

File: composer.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     ],
     "require": {
         "php": ">=7.1",
-        "psr/http-message": "^1.0",
+        "psr/http-message": "^1.0 || ^2.0",
```
Comment (Nyholm) [inline composer.json:19]: Do we need to write 1.1?

```suggestion
        "psr/http-message": "^1.1 || ^2.0",
```

**Review by Nyholm — APPROVED** [review]
Im happy with this. 

Just waiting on https://github.com/php-http/psr7-integration-tests/pull/68

File: composer.json [hunk 2]
```diff
@@ -16,13 +16,13 @@
...
         "php": ">=7.2",
-        "psr/http-message": "^1.0",
+        "psr/http-message": "^1.1 || ^2.0",
         "php-http/message-factory": "^1.0",
         "psr/http-factory": "^1.0"
     },
     "require-dev": {
         "phpunit/phpunit": "^7.5 || 8.5 || 9.4",
-        "php-http/psr7-integration-tests": "^1.0",
+        "php-http/psr7-integration-tests": "^1.0@dev",
```
Comment (nicolas-grekas) [inline composer.json:25]: This line looks desirable to me: testing with `@dev` for the test suite means anticipating any unreleased issue with them.

**Review by nicolas-grekas — COMMENTED** [review]
PR green, running with PSR-7 v2.

**Comment by Nyholm** [comment]
I reran the CI, it is all green. Let's merge

**Pull request merged**